test/e2e_env/universal/meshhealthcheck/policy.go

```diff
@@ -96,6 +96,115 @@ spec:
 		})
 	}, Ordered)
 
+	Describe("HTTP to real MeshService", func() {
+		meshName := "meshhealthcheck-http"
+		healthCheck := func(mesh, method, status string) string {
+			return fmt.Sprintf(`
+type: MeshHealthCheck
+mesh: %s
+name: everything-to-backend
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        name: test-server
+      default:
+        interval: 10s
+        timeout: 2s
+        unhealthyThreshold: 3
+        healthyThreshold: 1
+        failTrafficOnPanic: true
+        noTrafficInterval: 1s
+        healthyPanicThreshold: 0
+        reuseConnection: true
+        http: 
+          path: /%s
+          expectedStatuses: 
+          - %s`, mesh, method, status)
+		}
+
+		uniServiceYAML := fmt.Sprintf(`
+type: MeshService
+name: test-server
+mesh: %s
+labels:
+  kuma.io/origin: zone
+  kuma.io/env: universal
+spec:
+  selector:
+    dataplaneTags:
+      kuma.io/service: test-server
+  ports:
+  - port: 80
+    targetPort: 80
+    appProtocol: http
+`, meshName)
+		BeforeAll(func() {
+			err := NewClusterSetup().
+				Install(MeshUniversal(meshName)).
+				Install(YamlUniversal(healthCheck(meshName, "health", "200"))).
+				Install(DemoClientUniversal("dp-demo-client", meshName,
+					WithTransparentProxy(true)),
+				).
+				Install(TestServerUniversal("test-server", meshName, WithArgs([]string{"health-check", "http"}), WithProtocol(mesh.ProtocolHTTP))).
+				Install(YamlUniversal(uniServiceYAML)).
+				Install(YamlUniversal(`
```

Comment:
Don't we have a default hostname generator?

Reply:
I don't think so, we had it but it was reverted as e2e tests become flaky. I checked it now and I only see default HostnameGenerator for MES